### PR TITLE
[SWPWA-375] Terms and conditions support

### PR DESCRIPTION
**Pull request by dmitrijs-voronovs**
* Added terms and conditions to config store
* Added `TermsAndConditionsPopup` component
* Added support for terms and conditions on checkout step

Depends on https://github.com/scandipwa/store-graphql/pull/1

Screens:
Terms and conditions are Enabled:
Selection_080
Selection_081
* Popup with Terms and Conditions:
Selection_082

Terms and conditions are disabled:
Selection_083

### CODE REVIEW

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/CheckoutTermsAndConditionsPopup/CheckoutTermsAndConditionsPopup.component.js  1
- src/app/component/CheckoutTermsAndConditionsPopup/CheckoutTermsAndConditionsPopup.container.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4746780&bid=15838808)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/CheckoutTermsAndConditionsPopup/CheckoutTermsAndConditionsPopup.container.js  2
- src/app/component/CheckoutTermsAndConditionsPopup/CheckoutTermsAndConditionsPopup.component.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4746780&bid=15838808)

File: src/app/component/CheckoutBilling/CheckoutBilling.component.js [hunk 1]
```diff
@@ -22,27 +22,36 @@ import Field from 'Component/Field';
 
 import './CheckoutBilling.style';
```
Comment (alfredsgenkins) [inline src/app/component/CheckoutBilling/CheckoutBilling.component.js]: Please order imports. This should come last.

File: src/app/component/CheckoutBilling/CheckoutBilling.container.js [hunk 2]
```diff
@@ -13,32 +13,44 @@ import { PureComponent } from 'react';
...
-    totals: state.CartReducer.cartTotals
+    totals: state.CartReducer.cartTotals,
+    termsAreEnabled: state.ConfigReducer.terms_are_enabled,
+    termsAndConditions: state.ConfigReducer.checkoutAgreements
 });
 
 export const mapDispatchToProps = dispatch => ({
-    showErrorNotification: message => dispatch(showNotification('error', message))
+    showErrorNotification: message => dispatch(showNotification('error', message)),
+    showNewPopup: payload => dispatch(showPopup(TERMS_AND_CONDITIONS_POPUP_ID, payload))
```
Comment (alfredsgenkins) [inline src/app/component/CheckoutBilling/CheckoutBilling.container.js]: Why show `new` just show.

File: src/app/component/CheckoutBilling/CheckoutBilling.container.js [hunk 3]
```diff
@@ -113,6 +126,18 @@ export class CheckoutBillingContainer extends PureComponent {
         }
     }
 
+    showPopup() {
+        const { showNewPopup, termsAndConditions } = this.props;
+        const {
+            checkbox_text: title = 'Terms and Conditions',
+            content: text = 'There are no Terms and Conditions configured.'
```
Comment (alfredsgenkins) [inline src/app/component/CheckoutBilling/CheckoutBilling.container.js]: This must be translatable, no?

File: src/app/component/CheckoutBilling/CheckoutBilling.component.js [hunk 4]
```diff
@@ -51,10 +60,47 @@ class CheckoutBilling extends PureComponent {
...
+
+        return (
+            <>
+                <button
+                  block="CheckoutBilling"
+                  elem="Link"
+                  onClick={ this.handleShowPopup }
+                >
+                    { __('I agree to terms and conditions.') }
+                </button>
```
Comment (alfredsgenkins) [inline src/app/component/CheckoutBilling/CheckoutBilling.component.js]: Please do use button. Prefer label, this is not intuitive. Do it like:

I agree to [terms and conditions](http://openpopup.com) ☑️

Then on click on "terms and conditions" link - popup opens.

File: src/app/component/CheckoutBilling/CheckoutBilling.style.scss [hunk 5]
```diff
@@ -11,12 +11,26 @@
...
     }
+
+    &-Link {
+        &:hover {
+            text-decoration: underline;
+        }
+    }
+}
+
+.Button.CheckoutBilling {
```
Comment (alfredsgenkins) [inline src/app/component/CheckoutBilling/CheckoutBilling.style.scss]: Please do not do this. Find a different way.

File: src/app/store/Config/Config.reducer.js [hunk 6]
```diff
@@ -35,21 +35,27 @@ export const initialState = {
 };
 
 const ConfigReducer = (state = initialState, action) => {
-    const { config: { countries, reviewRatings, storeConfig = {} } = {}, type } = action;
+    const {
+        config: {
+            countries, reviewRatings, checkoutAgreements, storeConfig = {}
```
Comment (alfredsgenkins) [inline src/app/store/Config/Config.reducer.js]: Every on new line, please

File: src/app/query/Config.query.js [hunk 7]
```diff
@@ -17,6 +17,24 @@ export class ConfigQuery {
             .addFieldList(this._getStoreListFields());
     }
 
+    getCheckoutAgreements() {
+        return new Field('checkoutAgreements')
```
Comment (alfredsgenkins) [inline src/app/query/Config.query.js]: I have not seen this field implementation enywhere, is it default?

Comment (dmitrijs-voronovs) [reply src/app/query/Config.query.js]: Yes, it is provided by default

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/CheckoutTermsAndConditionsPopup/CheckoutTermsAndConditionsPopup.container.js  2
- src/app/component/CheckoutTermsAndConditionsPopup/CheckoutTermsAndConditionsPopup.component.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4746780&bid=15838808)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/CheckoutTermsAndConditionsPopup/CheckoutTermsAndConditionsPopup.container.js  2
- src/app/component/CheckoutTermsAndConditionsPopup/CheckoutTermsAndConditionsPopup.component.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ScandiPWA/base-theme/pullRequest?prid=4746780&bid=15838808)

**Comment by dmitrijs-voronovs** [comment]
Styling:
* desktop
Selection_129
* hover
Selection_130
* Mobile
Selection_131

**Pull request merged**